File: math/strong_number/python/strong_number.py
import math

sum1=0                                            
num=int(input("Enter a number:"))                # taking input from user
temp=num                                        # storing value of number in temporary variable as we need original later. 
while(num):                                     # loop to find the sum of "strong number" series
    i=1
    f=1
    r=num%10
    while(i<=r):
        f=f*i
        i=i+1
    sum1=sum1+f
    num=num//10
if(sum1==temp):                           # printing the result  
    print("The number is a strong number")
else:
    print("The number is not a strong number")
        
        
        
        
"""
Method 2:
Here i m going to add python in-built methods and smart code to deal this problem .
-sum(list):return total sum of element in the given list.
-math.factorial(int):return factorial of a number(+ve integer).
-int(str): type caste into integer if string pased.
"""        
def isStrongNum(number):
    return sum([math.factorial(int(i)) for i in str(number)])
        
if isStrongNum(num)==num:
        print("The number is a strong number,By Method2.")
else:
        print("The number is not a strong number,By Method 2.")

File: math/strong_number/Rust/strong_number.rs
fn check_strong_number(mut n: i32) -> bool {
    let temp = n;
    let mut sum = 0;
    while n > 0 {
        
        let r = n % 10;
        n /= 10;
        sum += fact(r);
    }
    if sum == temp {
        return true
    }
    false
}

fn fact(n :i32) -> i32 {
    match n {
        0 => 1,
        _ => n * fact(n-1)
    }
}

fn main() {
    let num = 145;
    if check_strong_number(num) {
        println!("{} is a strong number", num);
    }else {
        println!("{} is not a strong number", num);
    }
}